Improve scale jobs schedule to get better signal

shyamjvs opened the pull request:
Turns out our current schedule is less than optimal in terms of job timings. Changing it to fit in more runs, so that we have better signal coming from them.
Should help with spotting regressions easier.

The new schedule is as follows:

| Day | | |
| ------------- |:-------------:| -----:|
| Mon | 5k-node performance @ 00:01 PT | 2k-node performance @ 22:01 PT |
| Tue | 2k-node performance @ 05:01 PT | 5k-node correctness @ 14:01 PT |
| Wed | 5k-node performance @ 00:01 PT | 2k-node performance @ 22:01 PT |
| Thu | 2k-node performance @ 05:01 PT | 5k-node correctness @ 14:01 PT |
| Fri | 5k-node performance @ 00:01 PT | 2k-node performance @ 22:01 PT |
| Sat | 2k-node performance @ 05:01 PT | 5k-node correctness @ 14:01 PT |
| Sun | 'GKE' 2k-node performance @ 05:01 PT | 'GKE' 2k-node correctness @ 15:01 PT |

So:
- we'll have a performance run and a correctness run from each day (alternating b/w 2k and 5k though)
- we keep testing GKE (just like before) but exclusively on sunday
- the timings/timeouts have been set based on historical runs of each job
- we're experimenting with 5k-node correctness job running on weekend given that it's relatively stable now (can be changed to 2k later if it's causing trouble)

Also changed the cluster names for 2k jobs to be same as 5k, so that if one job failed to down the cluster, the following one will do that.

cc @kubernetes/sig-scalability-misc @porridge @krzyzacy @spiffxp

shyamjvs commented:
Also changed kubemark-100-high-density to run daily instead of weekly.

krzyzacy commented:
do you want people from your sig to review and stamp?

krzyzacy commented:
/lgtm
/hold
please fix timeout values in config.json (which need to be shorter than jenkins timeout)

shyamjvs commented:
Fixed it, thanks.
Would like @porridge to take a look before merging.

porridge commented:
In general, I love this change. I think increasing test frequency will make a big difference, allowing us to iterate much faster.
To visualize this, I copied this into [a calendar](https://calendar.google.com/calendar?cid=Z29vZ2xlLmNvbV9tNHA3bG1jODVubGlmazFxYzRnNTRqZjg4a0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) (see also attached screenshot for convenience) and I can see two issues / opportunities for improvement:
  1. since the first test that runs on Monday is longest, we don't get to see any results until Tuesday morning CET. Do you think it's feasible to rearrange the order such that we start for example with the shortest test (large-correctness), and a couple of hours earlier, such that we can see the first results around lunchtime on Monday CET?
  2. the durations on the calendar are the jenkins timeout. There is an over 1.5h overlap between consecutive tests. This sounds like too much - it could mean that a test which ran over could interfere with the next one, ruining both. Perhaps we just need to bring the various timeouts closer to the shortest one (OTOH I don't know their semantics. I guess that `jenkins_timeout` is when jenkins kills the top-most command, but what's the relation between `--timeout` and JSON `timeout`)?

![calendar](https://user-images.githubusercontent.com/489420/32878683-5c033e8e-caa7-11e7-9757-679f97f13e98.png)

porridge commented:
It would be great if you could update the [design doc](https://github.com/kubernetes/community/blob/master/contributors/devel/release/scalability-validation.md) once this is submitted.

shyamjvs commented:
Rebased. Thanks for creating the much-needed calendar @porridge. I'll update the design doc once this is merged.
Answering your questions:

- If you want results on monday morning you can look into the run from sunday night, right? If you're bothered that results of sunday are from GKE, I can swap the schedule of saturday and sunday to have GCE there instead. Though imo GKE should also be providing a good signal.
- There are 3 timeouts for each jenkins job and here's a summary of what they mean fwiu talking to @krzyzacy yesterday:
  - 'timeout' in config.json: This is the kubetest timeout (let's say X)
  - 'timeout' in bootstrap-ci.yaml: This is the whole job's timeout (i.e bootstrap.py in case of jenkins, and job pod in case of prow). It has to be at least (X+20) in order to do things like logs upload and cluster-down in those extra 20 mins. I changed it to 30 mins for our large jobs to give some extra time.
  - 'jenkins-timeout' in bootstrap-ci.yaml: This is the timeout offered by jenkins (traditionally set to X+120) but is not really used/needed anymore with the above timeout being the one used in practice. Besides, we're deprecating jenkins anyway.

So the one we really care about is the 2nd one (also please update the calendar with it), and that ensures no overlap across the jobs. For e.g you can see our kubemark-5000 runs timing out at 18h20m (https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-kubemark-gce-scale) which is 1100m - https://github.com/kubernetes/test-infra/blob/3918037ca80572eda660e81d3b6a43d97e04018b/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml#L168

shyamjvs commented:
Even after lgtm'ing, please keep the hold label on. I'll remove it and merge at a time such that we don't have conflict of a new run with a currently running.

porridge commented:
Thanks for explaining the timeouts. Calendar updated.
Regarding ordering, I'm not so sure about GKE, but let's see how it goes.
/lgtm

shyamjvs commented:
Merging this as there's no conflicting run currently.
/hold cancel

shyamjvs commented:
Sent out https://github.com/kubernetes/community/pull/1411 to update the design doc - PTAL.
